parasolr/solr/client.py

```diff
@@ -155,4 +217,13 @@ def query(self, wrap: bool = True, **kwargs: Any) -> Optional[QueryResponse]:
         response = self.make_request("post", url, headers=headers, params=kwargs)
         if response:
             # queries return the search response for now
-            return QueryResponse(response) if wrap else response
+
+            # unnless a raw/unwrapped result is requested,
+            # determine result type to use and initialize
+            if wrap:
+                result_class = QueryResponse
+                if "grouped" in response:
+                    result_class = GroupedResponse
+                response = result_class(response)
+
+            return response
```

Comment:
Nice, so we actually have 3 response types, sort of: QueryResponse (ungrouped), GroupedResponse, and unwrapped response object

Reply:
yes, that's right — and the unwrapped response is basically a `dict` (sometimes we need this for django templates because they call all callables, which does weird things with our attrdict object)

I toyed with creating a `DocumentList` object for the common part of the list of docs in a regular response and a grouped response, but it didn't seem like it actually got us anything — we're just accessing the fields, which we can already do